controllers/awsmachine_controller.go

```diff
@@ -323,11 +327,16 @@ func (r *AWSMachineReconciler) reconcileNormal(ctx context.Context, machineScope
 
 	switch instance.State {
 	case infrav1.InstanceStateRunning:
-		machineScope.Info("Machine instance is running", "instance-id", *machineScope.GetInstanceID())
+		machineScope.Info("EC2 instance is running", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
 		machineScope.SetReady()
 	case infrav1.InstanceStatePending:
-		machineScope.Info("Machine instance is pending", "instance-id", *machineScope.GetInstanceID())
+		machineScope.Info("EC2 instance is pending", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
+		r.Recorder.Eventf(machineScope.AWSMachine, corev1.EventTypeWarning, "InstancePending", "EC2 instance is %q", instance.State)
+	case infrav1.InstanceStateStopping, infrav1.InstanceStateStopped:
```

Comment:
This needs upstream support to make the Machine to flip back to `provisioned` state. Once https://github.com/kubernetes-sigs/cluster-api/issues/1622#issuecomment-545575154 is fixed

Reply:
If it is stopped, it's not guaranteed to have the same networkID when it returns up, so stopping/stopped should likely result in a permanent failure.

Reply:
Got it, so it should possibly transition to a failure state, like it did before.

@randomvariable what's the reason behind this change?

Reply:
what do you mean network ID? The IP address remains the same (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-lifecycle.html):
> The instance keeps its private IPv4 address. The instance gets a new public IPv4 address, unless it has an Elastic IP address, which doesn't change during a stop/start.

Reply:
In that case, the node should be able to come back up. Would you mind testing this use case on an actual cluster?

Reply:
Ah, right...  I had forgotten we aren't assigning public IPs :joy:

Do we have to worry about Node deletion with this, though? Doesn't the AWS cloud provider delete Nodes for instances that are in a stopped state?

Reply:
checking now.